### PR TITLE
Optionally implement std::error::Error for Error

### PR DESCRIPTION
**Pull request by qwandor**
This makes error handling easier when `std` is available, but is guarded behind a new feature `std`, to avoid pulling in the dependency for users who don't want it.

Also implemented `Display`, `Debug`, `Clone` and `Eq`.

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -9,8 +9,6 @@ categories = ["embedded", "hardware-support"]
 repository = "https://github.com/almusil/rfm69"
 description = "A generic rust driver to support RFM69 family wireless chips"
 
-# See more keys and their definitions at https://doc.rust-lang.org/cargo/reference/manifest.html
```
Comment (almusil) [inline Cargo.toml:12]: Unrelated change.

File: src/error.rs [hunk 2]
```diff
@@ -15,3 +17,19 @@ pub enum Error<Espi> {
     /// Packet exceeds maximum size (255 for send_large)
     PacketTooLarge,
 }
+
+impl<Espi: Display> Display for Error<Espi> {
```
Comment (almusil) [inline src/error.rs:21]: Display wasn't implemented for a reason, that's it significantly increases flash size as the formatting is pretty heavy weight (or at least it was last time when I was dealing with those issues).

**Review by almusil — CHANGES_REQUESTED** [review]
Hi, thank you for the PR I have two comments.

File: Cargo.toml (hunk 1 above)
Comment (qwandor) [reply Cargo.toml]: Reverted.

File: src/error.rs (hunk 2 above)
Comment (qwandor) [reply src/error.rs:21]: Using `Display` will increase code size, sure. But just having it implemented shouldn't be a problem, the code shouldn't be included in the binary if it is never called.

Comment (almusil) [reply src/error.rs:21]: It's true that the compiler should be smart enough to remove it when it's not referenced.

**Review by almusil — APPROVED** [review]
